handler/wipe.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"net/http"
+	"os"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+
+	"github.com/prometheus/pushgateway/storage"
+)
+
+func WipePersistentFile(
+	dms *storage.DiskMetricStore,
+	newStorage func() *storage.DiskMetricStore,
+	logger log.Logger) http.Handler {
+
+	// TODO: Should we return promhttp.InstrumentHandlerCounter and count calls to this endpoint?
+	return http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		level.Debug(logger).Log("msg", "wiping persistence file")
+		// Since we plan to wipe out the entire storage we actually don't care too much
+		// if there was an error persisting the last metrics, still I can see some value in logging
+		// potential errors for troubleshoting
+		if err := dms.Shutdown(); err != nil {
+			level.Error(logger).Log("msg", "problem shutting down metric storage", "err", err)
+		}
+
+		n := newStorage()
+		// Delete persitence file
+		// TODO: should we handle the error from os.Remove? What exactly should we be doing?
+		os.Remove(n.GetPersistenceFile())
```

Comment:
I think it is a bit low-level to ask the storage for a file to delete, delete it with `os.Remove` and then also rely on this code to create a new storage and not to keep using the old one with the deleted file.

How about giving the storage a `Wipe()` method that just does the stuff under the hood?

At some point, people might want to introduce storage implementation that use other backends than a file in a local file system.

Reply:
In fact, I think you shouldn't be concerned with deleting the file itself. (If the user hasn't requested persistence, there won't be any, but wiping of the (in-memory) storage should still be possible.)

The proposed `Wipe()` method should just iterate through everything there is in storage, delete it, and then let the usual persistence mechanism take care of writing that state to disk.

Reply:
Yeah actually it makes sense